### PR TITLE
Remove Arch-Type from strip command

### LINKED ISSUE #3950: docker image fails building for arm64
Was raised on the osm dev list (and is more a hunch as of now, but makes sense I think):

It might be not be possible to build our image on e.g. Apple Silicon. The error is caused by the name of the python bindings when we try to `strip` it:

https://github.com/valhalla/valhalla/blob/3fac9705794f902dceb9b031d04c2f4fa9ada109/Dockerfile#L45

I'd guess it builds actually `python_valhalla.cpython-310-darwin.so` or so.

### PR DESCRIPTION
**Pull request by homtec**
Fixes #3950

# Issue

What issue is this PR targeting? If there is no issue that addresses the problem, please open a corresponding issue and link it here.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,7 +42,7 @@ RUN tar -cvf valhalla.debug.tar valhalla_*.debug && gzip -9 valhalla.debug.tar
 RUN rm -f valhalla_*.debug
 RUN strip --strip-debug --strip-unneeded valhalla_* || true
 RUN strip /usr/local/lib/libvalhalla.a
-RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla.cpython-310-x86_64-linux-gnu.so
+RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla.cpython-310-*-gnu.so
```
Comment (nilsnolde) [inline Dockerfile:45]: tbh, not so sure about the `-gnu` either, but should be fine? anyways, better than before for sure.

**Review by nilsnolde — COMMENTED** [review]
Thanks @homtec ! Can you add a CHANGELOG entry?

File: Dockerfile (hunk 1 above)
Comment (kevinkreiser) [reply Dockerfile:45]: i agree, we can probably just do to be sure of ourselves

```suggestion
RUN strip /usr/lib/python3/dist-packages/valhalla/python_valhalla*.so
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 * **Removed**
    * REMOVED: Docker image pushes to Dockerhub [#4033](https://github.com/valhalla/valhalla/pull/4033)
 * **Bug Fix**
+   * FIXED: docker image fails building for arm64 [#3950](https://github.com/valhalla/valhalla/issues/3950)
```
Comment (nilsnolde) [inline CHANGELOG.md:5]: we organize our changelog in a date-descending manner, i.e. newest comes last. also we refer to the pull request, not the issue.

**Review by nilsnolde — CHANGES_REQUESTED** [review]

**Review by nilsnolde — APPROVED** [review]
thx:)

File: CHANGELOG.md (hunk 2 above)
Comment (nilsnolde) [reply CHANGELOG.md]: np;)

**Review by nilsnolde — APPROVED** [review]

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**